Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -467,6 +467,33 @@ class PurchasesOrchestratorTests: StoreKitConfigTestCase {
         }
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseSK2PackageWithInvalidPromotionalOfferSignatureThrowsError() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        self.customerInfoManager.stubbedCachedCustomerInfoResult = self.mockCustomerInfo
+        self.backend.stubbedPostReceiptResult = .success(self.mockCustomerInfo)
+
+        let product = try await self.fetchSk2Product()
+        let offer = PromotionalOffer.SignedData(
+            identifier: "identifier \(Int.random(in: 0..<1000))",
+            keyIdentifier: "key identifier \(Int.random(in: 0..<1000))",
+            nonce: .init(),
+            // This should be base64
+            signature: "signature \(Int.random(in: 0..<1000))",
```

Comment:
why not base64 the result of this so it looks realistic?

Reply:
This is 
> testPurchaseSK2PackageWithInvalidPromotionalOfferSignatureThrowsError

So I'm testing that if it's _not_ base64 (so it's just a random string), it produces the correct error instead of crashing or something.